moveit_setup_assistant/include/moveit/setup_assistant/tools/moveit_config_data.h

```diff
@@ -79,9 +79,9 @@ struct GroupMetaData
 };
 
 /**
- * Planning parameters which may be set in the config files
+ * Reusable parameter struct which may be used in reading and writing configuration files
  */
-struct OmplPlanningParameter
+struct GenericParameter
```

Comment:
this should actually be a class since it has member functions (see GStyle guide)

Reply:
Yes, i'll be working on that :+1:

Reply:
I think this struct is just passive.

I mean the member functions that uses this struct are `addGenericParameterToSensorPluginConfig()` and `SensorPluginConfig` is a member of the `moveit_config_data`, so if we made a class all it functions will be just `get` and `set` functions and the `addGenericParamereter` function would still be a function in the `movit_config_data` class.
Same with `getSensorPluginConfig()`.
Did I get it correctly?

Reply:
The controllers screen is also ready, but if this one really needs the struct to be class i'll modify it before I open a PR for it.

Reply:
I've made two commits, the first one uses a struct and in the second one I've converted it into a class.